tasks/collector.py

```diff
@@ -0,0 +1,108 @@
+import os
+import platform
+import subprocess
+import tempfile
+import urllib.request
+
+from invoke.exceptions import Exit
+from invoke.tasks import task
+
+from tasks.go import generate_licenses, tidy
+from tasks.libs.common.color import Color, color_message
+
+LICENSE_HEADER = """// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+"""
+OCB_VERSION = "v0.103.1"
+
+
+@task(post=[tidy, generate_licenses])
+def generate(ctx):
+    arch = platform.machine()
+    system = platform.system()
+    base_url = (
+        f"https://github.com/open-telemetry/opentelemetry-collector/releases/download/cmd%2Fbuilder%2F{OCB_VERSION}/"
+    )
+
+    if system == "Linux":
+        if arch == "x86_64":
+            binary_name = "ocb_0.103.1_linux_amd64"
+        elif arch == "arm64" or arch == "aarch64":
+            binary_name = "ocb_0.103.1_linux_arm64"
+        else:
+            print(f"Unsupported architecture: {arch}")
+            return
+    elif system == "Darwin":
+        if arch == "x86_64":
+            binary_name = "ocb_0.103.1_darwin_amd64"
+        elif arch == "arm64":
+            binary_name = "ocb_0.103.1_darwin_arm64"
+        else:
+            print(f"Unsupported architecture: {arch}")
+            return
+    else:
+        print(f"Unsupported system: {system}")
+        return
+
+    binary_url = f"{base_url}{binary_name}"
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        binary_path = os.path.join(tmpdirname, binary_name)
+        print(f"Downloading {binary_url} to {binary_path}...")
+
+        try:
+            urllib.request.urlretrieve(binary_url, binary_path)
+            os.chmod(binary_path, 0o755)
+            print(f"Downloaded to {binary_path}")
+        except Exception as e:
+            raise Exit(
+                color_message("Error: Failed to download the binary", Color.RED),
+                code=1,
+            ) from e
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should you retry the download on network errors?

Reply:
Have you consider building a custom runner that has the collector installed? The CI/CD team sooner or later will remove the possibility to download anything out of DD at CI time, the recommended way to have extra binaries is to have them in an image used by the ci runner